### DISCUSSION TITLE
Reserve the keyword it

### DISCUSSION THREAD

**Discussion by saulery** [vlang/v#11715] [Ideas]
Hello,

In a .map() context the `it` builtin variable can overlap an `it` variable at the function scope.

Wouldn't `it` be better to reserve the word it to avoid this?

Regards,

**Comment by ntrel** [comment]
That wouldn't solve the problem completely - there could be a `map` call nested inside another `map` call. It would be better if we could name the parameter for each `map`.

**Comment by saulery** [comment]
Hum, right. That adds more syntax sugar. I found the idea interesting to keep `it` even if it reserves a word rather than complicating the syntax to have a dynamic name for a trivial case.

And if .map() receive an anonymous function where the first parameter is the value of it?

I guess the same thing happens for special .sort() builtin variables.
